### ISSUE TITLE
[bug] expected struct Bug at line 10 column 49

### ISSUE THREAD

**Issue by shellscape** [moonrepo/moon#692]
**Describe the bug**

Moon doesn't appear to cover the various shapes of `package.json`

**Steps to reproduce**

Create some task and run it.
Create a project in the repo and add a package.json with the field:

```json
"bugs": "https://github.com/some/repo",
```

**Expected behavior**

No error

**Actual Behavior**

```
 ERROR 

Failed to parse ~/code/repo/packages/eslint-config/package.json: invalid type: string "https://github.com/some/repo/issues", expected struct Bug at line 10 column 49
```

**Screenshots**

**Environment**

```

```

**Additional context**

My prior gig was working with a company that consumed NPM updates in real time. Attempting to apply a strict schema to `package.json` is _a colossal mistake_ as NPM itself has no strict schema, and users do all kinds of insanely whacky things with package.json.

A `string` for that field, and others like `repository`, are completely valid. And I don't particularly want to go through all of my projects to update properties like that to their more verbose variant, just to satisfy moon. Please consider loosening the types and validation around this.

**Comment by shellscape** [comment]
For what it's worth, I've taken a look at https://github.com/moonrepo/moon/blob/master/crates/node/lang/src/package.rs#L16 and there are _a lot_ of incorrect assumptions about the shape of `package.json` in there. Happy to discuss further.

**Comment by milesj** [comment]
@shellscape Yeah this is definitely something we're looking to loosen. We've been slowly ripping pieces of this out because of this problem alone. I can patch this for this release today.

**Comment by shellscape** [comment]
Much appreciated. And offer stands of a call if you'd like to go over your typings and where the nuances with package.json lie. I have (unfortunately) a wealth of knowledge about how bananas package.json in the registry can be.

**Comment by milesj** [comment]
Went ahead and just ripped out a lot, heh: https://github.com/moonrepo/moon/pull/695

**Comment by shellscape** [comment]
Fixed by #695. Thanks again!
